eventuate-core/src/main/scala/com/rbmhtechnology/eventuate/EventsourcedProcessor.scala

```diff
@@ -186,9 +196,52 @@ trait EventsourcedProcessor extends EventsourcedWriter[Long, Long] {
       customDestinationAggregateIds = customDestinationAggregateIds,
       vectorTimestamp = lastVectorTimestamp,
       processId = DurableEvent.UndefinedLogId)
+
+  private def splitBatches(events: Vector[Seq[DurableEvent]]): (Vector[Seq[DurableEvent]], Vector[Seq[DurableEvent]]) = {
+    var num = 0
+    // Ensure that a single processEvent result
+    // is not partitioned across write batches.
+    events.span { w =>
+      val size = w.size
+      if (size > settings.writeBatchSize) {
+        throw new EventsourcedProcessingResultTooLargeException(size, settings.writeBatchSize)
+      } else {
+        num += size
+        num <= settings.writeBatchSize || num == size
```

Comment:
Right, this was a left-over from a version when the condition `(size > settings.writeBatchSize)` wasn't there yet. It should be there if we allowed batches larger than `writeBatchSize`